woocommerce/woocommerce-gutenberg-products-block
# Introduce Atomic Add to Cart Block

Adds an experimental/in-progress Add To Cart Atomic Block. This is a wrapper for the add to cart form button, qty picker, and any form elements that other product types require.

I've logged some inline todo's for context related follow ups (these are to support inner form elements and extensibility). There are also follow ups for:

- The variation picker (https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/2398)
- Grouped products (https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/2699)
- Extensibility (https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/2701)

So whilst the basic functionality is testable, this is not a final product :)

Closes #2676 

**How to test**

1. Create a fresh single product block
2. You should see the cart form! Test toggling the display of form elements in the inspector
3. Repeat this for each product type; simple, variation, variable, grouped
4. External product type should have no option for form elements
5. Test the button and form on the frontend for a simple product. The button should display, and qty picker should influance what is added to the cart.
